tasks/java.yml

```diff
@@ -1,7 +1,9 @@
 ---
 - name: "Check if Java is installed"
-  shell: which java
+  shell: command -v java
   register: check_java
+  failed_when: check_java.rc not in  (0, 1)
```

Comment:
Alright, so that looks like it's working fine.  I would expect an RC of 1 to mean failure.